### ISSUE TITLE
Add a REST API endpoint to change a user's password

### ISSUE THREAD

**Issue by john-thomas-dotcms** [dotCMS/core#19771]
**Is your feature request related to a problem? Please describe.**

Customer request, related to this ticket: https://dotcms.zendesk.com/agent/tickets/102882

Customers would like to be able to change user passwords, without logging into the back-end.  This is especially important for front-end users, whose entire interface with dotCMS is only through front-end pages.

Currently, there is no direct way to change a user's password without either logging into the back-end, or writing custom Java code (a plugin).

**Describe the solution you'd like**

Add a REST endpoint to initiate a password change for a user.  The endpoint should ideally:
* Require authentication
* For admin users, accept an optional user ID
* For non-admin users, or if no user ID is supplied, change the password for the currently authenticated user
* If no user ID is supplied, require specification of the current (old) password
* Require specification of the new password

**Describe alternatives you've considered**

Some customers have implemented OSGI plugins or other methods to perform password resets, but this requires custom code, which they then need to maintain across version upgrades.

To allow customers to fully manage front-end users without custom Java code, we need to provide REST endpoints which implement necessary functions such as password change.

**Comment by john-thomas-dotcms** [comment]
More generally, we need endpoints which provide all the user management functionality which is currently only available via the back-end UI or Java APIs.  This is a hole in our headless functionality.

Most important/needed:
* Create new users
* Change user password (this specific ticket request)
* Change user account information (email, etc.)

A couple others that might make sense as well (but I think we already have the ability to do this through other REST APIs, so these might not be necessary):
* Set permissions on objects for a given individual user
    * In other words, rather than setting individual user permissions on separate objects, specify a user account and give that account permissions to multiple objects - like we can do in the Users portlet in the BE UI.
* Assign user roles
